pkg/providers/cloudstack/template.go

```diff
@@ -236,6 +233,39 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, controlPlaneMachineSpec, etcd
 	return values, nil
 }
 
+func buildControlPlaneTemplate(controlPlaneMachineSpec *v1alpha1.CloudStackMachineConfigSpec, values map[string]interface{}) (content []byte, err error) {
+	cpTemplateName, ok := values[cpTemplateNameKey]
+	if !ok {
+		return nil, fmt.Errorf("unable to determine control plane template name")
+	}
+	cpMachineTemplate := MachineTemplate(fmt.Sprintf("%s", cpTemplateName), controlPlaneMachineSpec)
+	cpMachineTemplateBytes, err := templater.ObjectsToYaml(cpMachineTemplate)
+	if err != nil {
```

Comment:
The variables here are all prefixed with `cp`, but in this context there's only 1 template being rendered, the control plane, so we can simplify everything to suit the context of the function.